File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/FileAttachments.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' FileAttachments sample.
	''' </summary>
	Public Class FileAttachments
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		Public Shared Function Run(s1 As Stream, s2 As Stream) As SampleOutputInfo()
			Dim document As New PdfFixedDocument()
			document.DisplayMode = PdfDisplayMode.UseAttachments

			Dim helvetica As New PdfStandardFont(PdfStandardFontFace.Helvetica, 16)
			Dim blackBrush As New PdfBrush()
			Dim page As PdfPage = document.Pages.Add()
			page.Graphics.DrawString("This document contains 2 file attachments:", helvetica, blackBrush, 50, 50)
			page.Graphics.DrawString("1. fileattachments.cs.html", helvetica, blackBrush, 50, 70)
			page.Graphics.DrawString("2. fileattachments.vb.html", helvetica, blackBrush, 50, 90)

			Dim fileData1 As Byte() = New Byte(s1.Length - 1) {}
			s1.Read(fileData1, 0, fileData1.Length)
			Dim fileAttachment1 As New PdfDocumentFileAttachment()
			fileAttachment1.Payload = fileData1
			fileAttachment1.FileName = "fileattachments.cs.html"
			fileAttachment1.Description = "C# Source Code for FileAttachments sample"
			document.FileAttachments.Add(fileAttachment1)

			Dim fileData2 As Byte() = New Byte(s2.Length - 1) {}
			s2.Read(fileData2, 0, fileData2.Length)
			Dim fileAttachment2 As New PdfDocumentFileAttachment()
			fileAttachment2.Payload = fileData1
			fileAttachment2.FileName = "fileattachments.vb.html"
			fileAttachment2.Description = "VB.NET Source Code for FileAttachments sample"
			document.FileAttachments.Add(fileAttachment2)

			Dim output As SampleOutputInfo() = New SampleOutputInfo() {New SampleOutputInfo(document, "xfinium.pdf.sample.fileattachments.pdf")}
			Return output
		End Function
	End Class
End Namespace

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/FileAttachments.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// FileAttachments sample.
    /// </summary>
    public class FileAttachments
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        public static SampleOutputInfo[] Run(Stream s1, Stream s2)
        {
            PdfFixedDocument document = new PdfFixedDocument();
            document.DisplayMode = PdfDisplayMode.UseAttachments;

            PdfStandardFont helvetica = new PdfStandardFont(PdfStandardFontFace.Helvetica, 16);
            PdfBrush blackBrush = new PdfBrush();
            PdfPage page = document.Pages.Add();
            page.Graphics.DrawString("This document contains 2 file attachments:", helvetica, blackBrush, 50, 50);
            page.Graphics.DrawString("1. fileattachments.cs.html", helvetica, blackBrush, 50, 70);
            page.Graphics.DrawString("2. fileattachments.vb.html", helvetica, blackBrush, 50, 90);

            byte[] fileData1 = new byte[s1.Length];
            s1.Read(fileData1, 0, fileData1.Length);
            PdfDocumentFileAttachment fileAttachment1 = new PdfDocumentFileAttachment();
            fileAttachment1.Payload = fileData1;
            fileAttachment1.FileName = "fileattachments.cs.html";
            fileAttachment1.Description = "C# Source Code for FileAttachments sample";
            document.FileAttachments.Add(fileAttachment1);

            byte[] fileData2 = new byte[s2.Length];
            s2.Read(fileData2, 0, fileData2.Length);
            PdfDocumentFileAttachment fileAttachment2 = new PdfDocumentFileAttachment();
            fileAttachment2.Payload = fileData1;
            fileAttachment2.FileName = "fileattachments.vb.html";
            fileAttachment2.Description = "VB.NET Source Code for FileAttachments sample";
            document.FileAttachments.Add(fileAttachment2);

            SampleOutputInfo[] output = new SampleOutputInfo[] { new SampleOutputInfo(document, "xfinium.pdf.sample.fileattachments.pdf") };
            return output;
        }
    }
}